dev/puppet-server.conf.sample

```diff
@@ -65,6 +65,18 @@ http-client: {
     #                TLS_RSA_WITH_AES_256_CBC_SHA,
     #                TLS_RSA_WITH_AES_128_CBC_SHA256,
     #                TLS_RSA_WITH_AES_128_CBC_SHA]
+
+    # The amount of time, in milliseconds, that an outbound HTTP connection
+    # will wait for no data to be available before closing the socket. If not
+    # defined, defaults to 20 minutes. If 0, the timeout is infinite and if
+    # negative, the value is undefined (system default).
```

Comment:
> "is undefined in the application, the behavior is governed by the system default behavior"

Yeah, that sounds like an improvement and accurate.

> Are we talking about a socket in the connected state, or are we talking about the timeout associated with establishing a TCP connection before it enters the connected state?

The `socket-timeout` would apply to a socket in the connected state.  If the value were set to 0 and, assuming the underlying network layers were cooperating, the socket would persist indefinitely.  The "20 minute" value is a defensive measure to guard against the case that the server otherwise might leave the socket open indefinitely as well.  We saw this to be the case with a thundering herd testing scenario where Puppet Server was connected to a PuppetDB which was essentially hanging the connections indefinitely because it was unable to write data to the disk.  20 minutes ideally should be way longer than a reasonable client request should take.